bagitobjecttransfer/recordtransfer/models.py

```diff
@@ -39,138 +42,364 @@ def get_full_name(self):
+        """Start a new upload session."""
         return cls(token=get_random_string(length=32), started_at=timezone.now())
 
     @property
-    def upload_size(self):
-        """Get total size (in bytes) of all uploaded files in this session"""
-        return sum(f.file_upload.size for f in self.get_existing_file_set())
+    def upload_size(self) -> int:
+        """Get total size (in bytes) of all uploaded files in this session. This includes the size
+        of both temporary and permanent files.
+        """
+        if self.status in (
+            self.SessionStatus.EXPIRED,
+            self.SessionStatus.DELETED,
+        ):
+            raise ValueError(
+                f"""Cannot get upload size from session {self.token} because the session has
+                expired or has been deleted"""
+            )
+        elif self.status in (
+            self.SessionStatus.COPYING_IN_PROGRESS,
+            self.SessionStatus.REMOVING_IN_PROGRESS,
+        ):
+            raise ValueError(
+                f"""Cannot get upload size from session {self.token} while copying or removing
+                files is in progress"""
+            )
+        return sum(
+            f.file_upload.size
+            for f in chain(self.tempuploadedfile_set.all(), self.permuploadedfile_set.all())
+            if f.exists
+        )
+
+    @property
+    def file_count(self) -> int:
+        """Get the total count of temporary + permanent uploaded files."""
+        if self.status in (
+            self.SessionStatus.EXPIRED,
+            self.SessionStatus.DELETED,
+        ):
+            raise ValueError(
+                f"""Cannot get file count from session {self.token} because the session has
+                expired or has been deleted"""
+            )
+        elif self.status in (
+            self.SessionStatus.COPYING_IN_PROGRESS,
+            self.SessionStatus.REMOVING_IN_PROGRESS,
+        ):
+            raise ValueError(
+                f"""Cannot get file count from session {self.token} while copying or removing files
+                is in progress"""
+            )
+        return sum(
+            f.exists
+            for f in chain(self.permuploadedfile_set.all(), self.tempuploadedfile_set.all())
+        )
 
-    def get_existing_file_set(self):
-        """Get all files from the uploadedfile_set where the file exists"""
-        return [f for f in self.uploadedfile_set.all() if f.exists]
+    def add_temp_file(self, file: UploadedFile) -> "TempUploadedFile":
+        """Add a temporary uploaded file to this session."""
+        if self.status not in (self.SessionStatus.CREATED, self.SessionStatus.UPLOADING):
+            raise ValueError(
+                f"""Cannot add temporary uploaded file to session {self.token} because the session
+                status is {self.status} and not {self.SessionStatus.CREATED} or
+                {self.SessionStatus.UPLOADING}"""
+            )
```

Comment:
Similar to my other suggestions, """ should be changed to ", but I'm also suggesting you use the names from the enum rather than their values.

This has the effect of outputting "... and not CREATED or UPLOADING" versus "... and not CR or UG". The human readable names should be preferred in error messages.

```suggestion
            raise ValueError(
                f"Cannot add temporary uploaded file to session {self.token} because the session "
                f"status is {self.status.name} and not {self.SessionStatus.CREATED.name} or "
                f"{self.SessionStatus.UPLOADING.name}"
            )
```

Reply:
Addressed by: https://github.com/NationalCentreTruthReconciliation/Secure-Record-Transfer/pull/318/commits/f216239bd7f9c5e521579aaa15340758e0bcb71f